### PR TITLE
txn: skip the transaction start_ts used by analyze calculating the gc

### LINKED ISSUE #35062: Long auto-analyze txn blocks GC
## Enhancement

After #32726, the auto-analyze txns will not be interrupted by GC, instead, it blocks GC. The auto-analyzes will not repeatedly fail by blocking GC, but it brings some performance degradation when GC is blocked for a long time.

Either make auto-analyze fast enough or not execute auto-analyze it in a txn can improve this case.

### PR DESCRIPTION
**Pull request by cfzjywxk**
### What problem does this PR solve?

Issue Number: close #35062 

Problem Summary:

Back pick https://github.com/pingcap/tidb/pull/35144 to the master branch as it's urgent to fix the issue 
on the release-6.1 branch. This is a workaround solution, it's still needed to solve it completely in the 
next release.

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: session/session.go [hunk 1]
```diff
@@ -3240,12 +3240,16 @@ func (s *session) ShowProcess() *util.ProcessInfo {
...
+	var startTS, processInfoID uint64
 	tmp, ok := se.(*session)
 	if !ok {
 		logutil.BgLogger().Error("GetStartTSFromSession failed, can't transform to session struct")
-		return 0
+		return 0, 0
+	}
+	processInfo := tmp.ShowProcess()
+	if processInfo != nil {
+		processInfoID = processInfo.ID
```
Comment (TonsnakeLin) [inline session/session.go:3252]: we can get session id from tmp.sessionvars.id
the rest LGTM

**Review by TonsnakeLin — APPROVED** [review]

File: session/session.go (hunk 1 above)
Comment (cfzjywxk) [reply session/session.go:3252]: I think it's safer to distinguish the `processid` and `sessionid`, comparing the `processid` explicitly avoids the implicit constraint that they must be the same as the `analyze` process injects the `processid` at the start.

**Review by MyonKeminta — APPROVED** [review]

**Pull request merged**